src/art/utils/deploy_model.py

```diff
@@ -0,0 +1,115 @@
+import asyncio
+import json
+import os
+import time
+import aiohttp
+from art.model import TrainableModel
+
+
+def init_together_session() -> aiohttp.ClientSession:
+    """
+    Initializes a session for interacting with Together.
+    """
+    if "TOGETHER_API_KEY" not in os.environ:
+        raise ValueError("TOGETHER_API_KEY is not set, cannot deploy LoRA to Together")
+    session = aiohttp.ClientSession()
+    session.headers.update(
+        {
+            "Authorization": f"Bearer {os.environ['TOGETHER_API_KEY']}",
+            "Content-Type": "application/json",
+        }
+    )
+    return session
+
+
+def model_checkpoint_id(model: TrainableModel, step: int) -> str:
+    """
+    Generates a unique ID for a model checkpoint.
+    """
+    return f"{model.project}-{model.name}-{step}"
+
+
+async def previously_deployed_model_id(model: TrainableModel) -> str | None:
+    """
+    Checks if a model with the same name has already been deployed to Together.
+    If so, returns the model ID.
+    """
+    async with init_together_session() as session:
+        async with session.get(url="https://api.together.xyz/v1/models") as response:
+            response.raise_for_status()
+            result = await response.json()
+
+            # find a model with an "id" that contains the model.name
+            for deployed_model in result:
+                if model.name in deployed_model["id"]:
+                    return deployed_model["id"]
+
+            return None
+
+
+async def deploy_together(
```

Comment:
Can we add an explicit check in this function that the base model is one that Together supports? I guess the user will figure it out eventually if they try one that isn't supported but would be nice to have it explicitly up front so you can take that into account when deciding which base model to train.

Reply:
Now returning a `UnsupportedBaseModelDeploymentError` error.